File: android/app/src/main/java/com/coinswallet/coins/ThorCoin.java
// EthereumCoin.java
package com.coinswallet.coins;

import com.coinswallet.CoinFactory;

import wallet.core.jni.Base64;
import wallet.core.jni.CoinType;
import wallet.core.jni.PrivateKey;

public class ThorCoin extends CoinFactory.Coin {
    private final String address;
    private final PrivateKey privateKey;


    public ThorCoin(String mnemonic) {
        super(mnemonic);
        this.address = CoinType.THORCHAIN.deriveAddress(wallet.getKeyForCoin(CoinType.THORCHAIN));
        this.privateKey = wallet.getKeyForCoin(CoinType.THORCHAIN);
    }

    @Override
    public String getNewAddress(Boolean isTestNet) {
        return this.address;
    }

    @Override
    public String getPrivateKey(Boolean isTestNet) {
        return Base64.encode(this.privateKey.data());
    }

    @Override
    public String signTransaction(String rawData) {
        return null;
    }
}

File: ios/ThorCoin.swift
//
//  EthereumCoin.swift
//  coinswallet
//
//  Created by Tal Grynbaum on 30/06/2023.
//

import Foundation
import WalletCore

class ThorCoin: CoinFactory.Coin {
    private var address: String = ""

    override init(mnemonic: String) {
        super.init(mnemonic: mnemonic)
      self.address = CoinType.thorchain.deriveAddress(privateKey: wallet.getKeyForCoin(coin: .thorchain))
    }
  

    override func getNewAddress(isTestNet:Bool) -> String {
        return self.address
    }

    override func getPrivateKey(isTestNet:Bool) -> String {
        let privateKeyBytes = wallet.getKeyForCoin(coin: .thorchain).data
        return Base64.encode(data: privateKeyBytes)
    }

    override func signTransaction(rawData: String) -> String {
        // Implement the logic to sign an Ethereum transaction using the provided raw data
        return ""
    }
}
